server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -223,6 +223,13 @@ public CopyTo copyTo() {
         return copyTo;
     }
 
+    /**
+     * A value to use in place of a {@code null} value in the document source.
+     */
+    protected Object nullValue() {
```

Comment:
How bad would it be to make this `abstract` so implemeneters need to think about it?

Reply:
To me, `null_value` isn't a cross-cutting feature: only 9 out of ~40 field mappers implement it. In many cases it doesn't make that much sense (or could be harmful?), like setting a `null_value` for a geopoint. So I'm inclined to avoid making it abstract and encouraging mappers to implement it.